cmd/tealdbg/debugger.go

```diff
@@ -209,21 +293,30 @@ func (s *session) Resume() {
 }
 
 // setBreakpoint must be called with lock taken
+// Used for setting a breakpoint in step execution and adding bp to the session.
 func (s *session) setBreakpoint(line int) error {
 	if line >= len(s.breakpoints) {
 		return fmt.Errorf("invalid bp line %d", line)
 	}
 	s.breakpoints[line] = breakpoint{set: true, active: true}
-	s.debugConfig = debugConfig{BreakAtLine: breakpointLine(line)}
+	s.debugConfig.setActiveBreak(line)
 	return nil
 }
 
 func (s *session) SetBreakpoint(line int) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	// Reset all existing flags and breakpoints and set a new bp.
```

Comment:
why moved the resetting `s.debugConfig` from `SetBreakpoint` to `setBreakpoint`?

Reply:
Good catch - I will move this back into `setBreakpoint`

Reply:
Actually, for `Resume` there can be multiple active breakpoints (code execution can jump backward and break on a previous bp) set so I moved this outside instead - let me know if that makes sense.